internal/storage/file.go

```diff
@@ -0,0 +1,89 @@
+package storage
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+)
+
+func NewFileStorage(fileName string) (FileStorage, error) {
+	storage := FileStorage{
+		file:  fileName,
+		vault: map[string]string{},
+	}
+	file, err := os.OpenFile(fileName, os.O_RDONLY|os.O_CREATE, 0777)
+	if err != nil {
+		return storage, err
+	}
+	defer file.Close()
+
+	decoder := json.NewDecoder(file)
+	for {
+		var link Link
+		err := decoder.Decode(&link)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return storage, err
+		}
```

Comment:
все гуд, просто может такая запись могла бы быть менее перегруженной:
```suggestion
                 if err == io.EOF {
			break
		}
		if err != nil {
			return storage, err
		}
```